### PR TITLE
Doc alter source add column

### LINKED ISSUE #1207: Document: feat: support alter source add column
This issue tracks the documentation update needed for the merged PR #11350.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/11350
Source PR Merged At: 2023-08-24T08:07:09Z

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  Document ADD COLUMN clause for ALTER SOURCE

- **Notes**

  [ Include any supplementary context or references here. ]

- **Related code PR**

  https://github.com/risingwavelabs/risingwave/pull/11350

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1207

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Review by wugouzi — APPROVED** [review]
LGTM

File: docs/sql/commands/sql-alter-source.md [hunk 1]
```diff
@@ -5,13 +5,13 @@ description: Modify existing source name.
...
 
-Use the `ALTER SOURCE` command along with `RENAME TO` to modify the existing source name.
+Use the `ALTER SOURCE` command to add columns to the source or modify the name of the source. If your source was created with a schema registry, columns cannot be altered.
 
 ## Syntax
 
 ```sql
-ALTER SOURCE current_source_name
-   RENAME TO new_source_name;
+ALTER SOURCE current_source_name ADD COLUMN col_name data_type
```
Comment (hengm3467) [inline docs/sql/commands/sql-alter-source.md]: ADD COLUMN should also be optional: [ ADD COLUMN col_name data_type ]

File: docs/sql/commands/sql-alter-source.md [hunk 2]
```diff
@@ -5,13 +5,13 @@ description: Modify existing source name.
...
+    [ RENAME TO new_source_name ];
 ```
 
 import rr from '@theme/RailroadDiagram'
```
Comment (hengm3467) [inline docs/sql/commands/sql-alter-source.md]: We can remove the diagram now.

File: docs/sql/commands/sql-alter-source.md (hunk 1 above)
Comment (hengm3467) [reply docs/sql/commands/sql-alter-source.md]: But at least one of "ADD COLUMN" and "RENAME TO" should be specified.

**Pull request merged**